pkg/compose/build_buildkit.go

```diff
@@ -76,3 +84,23 @@ func (s *composeService) doBuildBuildkit(ctx context.Context, opts map[string]bu
 
 	return imagesBuilt, err
 }
+
+func (s composeService) dryRunBuildResponse(ctx context.Context, options map[string]build.Options) (resp map[string]*client.SolveResponse, err error) {
+	buildResponse := map[string]*client.SolveResponse{}
+	for name, option := range options {
+		dryRunUUID := fmt.Sprintf("dryRun-%x", sha1.Sum([]byte(name)))
+		args := []string{"buildx", "build", "--print=outline", "--output=type=cacheonly", "-f", option.Inputs.DockerfilePath, option.Inputs.ContextPath}
```

Comment:
Sweet, but different from https://github.com/docker/compose/blob/e606701d9f7272e1858a4ce9d189e5347f337c70/pkg/compose/build_buildkit.go#L56 which is vendored. 

Not the same build options being passed.